From: sohee_yoon <sohee_yoon@tmax.co.kr>
Date: Tue, 6 Apr 2021 11:25:26 +0900
Subject: [feat] Arc Menu : change icon size of menu button icon

- change icon size of menu button icon
- reposition menu button

[IMS] N/A
---
 extensions/arc-menu/controller.js  | 2 +-
 extensions/arc-menu/menuButton.js  | 2 ++
 extensions/arc-menu/menuWidgets.js | 4 ++--
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/extensions/arc-menu/controller.js b/extensions/arc-menu/controller.js
index b50c6c8..260dd7b 100644
--- a/extensions/arc-menu/controller.js
+++ b/extensions/arc-menu/controller.js
@@ -264,7 +264,7 @@ var MenuSettingsController = class { // eslint-disable-line no-unused-vars
     _setButtonIconSize() {
         let menuButtonWidget = this._menuButton.menuButtonWidget;
         let stIcon = menuButtonWidget.getPanelIcon();
-        stIcon.icon_size = 25;
+        stIcon.icon_size = 43;
     }
 
     _setButtonIconPadding() {
diff --git a/extensions/arc-menu/menuButton.js b/extensions/arc-menu/menuButton.js
index 9f7ca20..443ca53 100644
--- a/extensions/arc-menu/menuButton.js
+++ b/extensions/arc-menu/menuButton.js
@@ -73,6 +73,8 @@ var MenuButton = GObject.registerClass(class ArcMenuButton extends PanelMenu.But
         this.y_expand = false;
 
         // Add Menu Button Widget to Button
+        this.container.style = 'width: 50px;';
+        this.remove_style_class_name('panel-button');
         this.add_actor(this.menuButtonWidget.actor);
     }
 
diff --git a/extensions/arc-menu/menuWidgets.js b/extensions/arc-menu/menuWidgets.js
index cabd57a..9c220e2 100644
--- a/extensions/arc-menu/menuWidgets.js
+++ b/extensions/arc-menu/menuWidgets.js
@@ -1909,9 +1909,10 @@ var PlaceMenuItem = GObject.registerClass(class ArcMenuPlaceMenuItem extends Arc
 var MenuButtonWidget = class ArcMenuMenuButtonWidget { // eslint-disable-line no-unused-vars
     constructor() {
         this.actor = new St.BoxLayout({
-            style_class: 'panel-status-menu-box',
             pack_start: false,
+            x_align: Clutter.ActorAlign.END,
         });
+        this.actor.style = 'width: 45px;';
         this._arrowIcon = PopupMenu.arrowIcon(St.Side.BOTTOM);
         this._arrowIcon.add_style_class_name('arc-menu-arrow');
 
@@ -1931,7 +1932,6 @@ var MenuButtonWidget = class ArcMenuMenuButtonWidget { // eslint-disable-line no
         this.actor.add_child(this._icon);
         this.actor.add_child(this._label);
         this.actor.add_child(this._arrowIcon);
-
     }
 
     setActiveStylePseudoClass(enable) {
